ducktape/cluster/remoteaccount.py

```diff
@@ -30,7 +61,99 @@ def __init__(self, hostname, user=None, ssh_args=None, ssh_hostname=None, extern
         self.ssh_args = ssh_args
         self.ssh_hostname = ssh_hostname
         self.externally_routable_ip = externally_routable_ip
-        self.logger = logger
+        self._logger = logger
+        self._ssh_config = None
+        self._ssh_client = None
+        self._sftp_client = None
+
+    @property
+    def logger(self):
+        if self._logger:
+            return self._logger
+        else:
+            return logging.getLogger(__name__)
+
+    @logger.setter
+    def logger(self, logger):
+        self._logger = logger
+
+    def _log(self, level, msg, *args, **kwargs):
+        msg = "%s: %s" % (str(self), msg)
+        self.logger.log(level, msg, *args, **kwargs)
+
+    @property
+    def ssh_config(self):
+        if not self._ssh_config:
+            self._ssh_config = self._parse_ssh_opts()
+        return self._ssh_config
+
+    @property
+    def ssh_client(self):
+        if not self._ssh_client:
+            o = self.ssh_config.lookup(self.hostname)
+
+            client = SSHClient()
+            client.set_missing_host_key_policy(AutoAddPolicy())
```

Comment:
Do we actually want to add it? Adding host keys for ec2 is actually kind of annoying because over time they collect and since ec2 will recycle the host names over time, eventually you get host key conflicts. I put this in my `~/.ssh/config` to avoid this:

```
Host *.compute.amazonaws.com
  StrictHostKeyChecking no
  UserKnownHostsFile /dev/null
  LogLevel QUIET
```

A quick look suggests paramiko doesn't have a policy that just accepts the host key but doesn't add it, but maybe we could just implement a new `MissingHostKeyPolicy` that does that?